### ISSUE TITLE
bug(react/Menu): Hitting enter on MenuButton does not trigger the menu to open

### ISSUE THREAD

**Issue by hbuchel** [aws-amplify/amplify-ui#5310]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [x] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Primitive components

### How is your app built?

Next.js

### Please describe your bug.

If you hit enter on any of the [Menu elements ](https://ui.docs.amplify.aws/react/components/menu) it does not trigger the open functionality. In the first example, it triggers the onClick from the first MenuItem, `alert('Download')`. The MenuButton should be toggle-able with space or enter https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/#keyboard_interaction

The Radix component used as the base for this component works correctly https://www.radix-ui.com/primitives/docs/components/dropdown-menu#accessibility

### What's the expected behaviour?

Hitting enter on the MenuButton should toggle the menu.

### Help us reproduce the bug!

Hit enter on any of the menu buttons on the Docs page for Menu https://ui.docs.amplify.aws/react/components/menu

### Code Snippet

```typescript
// Put your code below this line.

```

**Comment by aadimch** [comment]
This issue was fixed after a change to the underlying dependency(radix-ui). Verified that it is working as intended on the docs site [here](https://ui.docs.amplify.aws/react/components/menu).
